### PR TITLE
add GetArchiveHoursMul up/down command

### PR DESCRIPTION
**Pull request by DeadByElpy**

### CODE REVIEW

File: src/CommandBinaryBuffer.ts [hunk 1]
```diff
@@ -19,6 +25,46 @@ export interface IBatteryVoltage {
...
+
+
+    /**
+     * Value time.
+     */
+    time: Seconds | undefined,
+
+
+    /**
+     * Normal da
```
Comment (DarkPark) [inline src/CommandBinaryBuffer.ts:62]: misprint?

File: src/CommandBinaryBuffer.ts [hunk 2]
```diff
@@ -184,7 +230,14 @@ class CommandBinaryBuffer extends BinaryBuffer {
     }
 
     setHours ( hour: number, hours: number ): void {
-        this.setUint8(((hours & 0x07) << 5) | (hour & 0x1f));
+        let hoursAmount = hours;
```
Comment (DarkPark) [inline src/CommandBinaryBuffer.ts:233]: in different places there are `hoursAmount` and `hourAmount`

File: src/CommandBinaryBuffer.ts [hunk 3]
```diff
@@ -236,6 +289,60 @@ class CommandBinaryBuffer extends BinaryBuffer {
 
         [lowVoltageByte, lowAndHighVoltageByte, highVoltageByte].forEach(byte => this.setUint8(byte));
     }
+
+    getChannelsValuesWithHourDiff (): {hourAmount: number, channels: Array<IChannel>, date: Date} {
```
Comment (DarkPark) [inline src/CommandBinaryBuffer.ts:293]:
```suggestion
    getChannelsValuesWithHourDiff (): ISomething {
```

?

File: src/commands/uplink/GetArchiveHoursMul.ts [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+const COMMAND_TITLE = 'GET_ARCHIVE_HOURS_MUL';
+
+// date 2 bytes, hour 1 byte, channels - 1 byte, so max channels = 4
+// max hours diff - 7 (3 bit value)
+// 4 + (4 channels * 5 bytes of hour values) + (4 * 5 bytes of diff * 7 max hours diff)
+const COMMAND_BODY_MAX_SIZE = 164;
+
+
+class GetArchiveHoursMul extends GetCurrentMul {
+    constructor ( public parameters: {channels: Array<IChannel>, date: Date | undefined | string, hourAmount: number | undefined} ) {
```
Comment (DarkPark) [inline src/commands/uplink/GetArchiveHoursMul.ts:18]: interface for parameters?

File: src/CommandBinaryBuffer.ts (hunk 2 above)
Comment (DeadByElpy) [reply src/CommandBinaryBuffer.ts]: fixed

File: src/CommandBinaryBuffer.ts (hunk 3 above)
Comment (DeadByElpy) [reply src/CommandBinaryBuffer.ts:293]: no idea, let's keep it like that for now

File: src/commands/uplink/GetArchiveHoursMul.ts (hunk 4 above)
Comment (DeadByElpy) [reply src/commands/uplink/GetArchiveHoursMul.ts]: fixed

File: src/commands/downlink/GetArchiveHoursMul.ts [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+ * GetArchiveHoursMul command parameters
+ *
+ * @example
+ * // time: 2023-04-03T14:01:17.000Z
+ * {sequenceNumber: 77, time: 733845677}
+ */
+interface IDownlinkGetArchiveHoursMulParameters {
+    /** amount of hours to retrieve */
+    hourAmount: number,
+    /** seconds */
```
[the remark this answers is not in the document]
Comment (DeadByElpy) [reply src/commands/downlink/GetArchiveHoursMul.ts]: fixed

File: src/commands/downlink/GetArchiveHoursMul.ts [hunk 6]
```diff
@@ -0,0 +1,103 @@
...
+import {DIRECTION_TYPE_DOWNLINK} from '../../constants/directionTypes.js';
+import {getSecondsFromDate, getDateFromSeconds} from '../../utils/time.js';
+
+
+/**
+ * GetArchiveHoursMul command parameters
+ *
+ * @example
+ * // time: 2023-04-03T14:01:17.000Z
+ * {sequenceNumber: 77, time: 733845677}
```
[the remark this answers is not in the document]
Comment (DeadByElpy) [reply src/commands/downlink/GetArchiveHoursMul.ts]: fixed

**Review by DarkPark — APPROVED** [review]

**Pull request merged**